src/components/Payment/Payment.tsx

```diff
@@ -70,6 +70,8 @@ const Payment = ({
         return t('user:payment.daily_subscription');
       case 'week':
         return t('user:payment.weekly_subscription');
+      case 'granted':
+        return t('user:payment.granted_subscription');
       default:
         throw 'Unknown period';
```

Comment:
Maybe it's worth it to return a generic placeholder here instead of throwing an exception? That way the page won't die if there are any other edge cases. Maybe just 'Other' would be OK?

Reply:
Wanted to make as less changes as possible so I did not touched it, but I guess it makes sense.